service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -32,15 +52,23 @@ default void bindServices(ServiceBinding<?>... serviceBindings) {
             }
         }
 
+        ServiceBinding<?> primaryServiceBinding = serviceBindings[0];
         // Bind the service info for the first one passed in
-        binder.bind(ServiceInfo.class).toProvider(new ServiceInfoProvider(serviceBindings[0].serviceInterface()));
+        binder.bind(ServiceInfo.class).toProvider(
+                new ServiceInfoProvider(
+                        primaryServiceBinding.serviceInterface(),
+                        Arrays
+                                .stream(serviceBindings)
+                                .map(ServiceBinding::serviceInterface)
+                                .toArray(Class[]::new)
+                ));
```

Comment:
This is a good point, and maybe a bit of an issue. Let's think about it from the client side: let's say I want to consume the metrics from another service. How would I do it?

This is making me think that we shouldn't be registering everything as `Descriptor.name()` but maybe as something like `s"${primaryServiceDesciptor.name}/${descriptor.name}"`

Unfortunately, I think this extends out to the client interface as well. There's no way to say that I want a client of the user service's metrics service vs the item service's metrics service. This makes the entire idea of shared service interfaces flawed.

Also, I think the confusion between gateway proxying and service location has crept back in.